### PR TITLE
[irods/irods#7567] Deprecate `--link`, add `--ignore-symlinks` to `iput` and `irsync` (main)

### PR DESCRIPTION
**Pull request by korydraughn**
Companion PR: irods/irods#7567

Not yet tested.

### CODE REVIEW

File: src/iput.cpp [hunk 1]
```diff
@@ -193,7 +193,8 @@ usage( FILE* _fout ) {
         "       it in the catalog.",
         " -K  verify checksum - calculate and verify the checksum on the data, both",
         "       client-side and server-side, and store it in the catalog.",
-        " --link - ignore symlink.",
+        " --link - [Deprecated] ignore symlink. Use --ignore-symlinks.",
```
Comment (trel) [inline src/iput.cpp:196]: plural please

Comment (korydraughn) [reply src/iput.cpp:196]: Dang, I missed another one.

Will do.

Comment (korydraughn) [reply src/iput.cpp]: Done.

**Review by alanking — COMMENTED** [review]
No complaints here. Squash it

File: src/iput.cpp [hunk 2]
```diff
@@ -193,7 +193,8 @@ usage( FILE* _fout ) {
         "       it in the catalog.",
         " -K  verify checksum - calculate and verify the checksum on the data, both",
         "       client-side and server-side, and store it in the catalog.",
-        " --link - ignore symlink.",
+        " --link - [Deprecated] ignore symlinks. Use --ignore-symlinks.",
+        " --ignore-links - ignore symlinks.",
```
Comment (korydraughn) [inline src/iput.cpp:197]: Should be `--ignore-symlinks`

**Review by alanking — COMMENTED** [review]
I'm fine with linking #433

File: src/iput.cpp [hunk 3]
```diff
@@ -238,5 +238,6 @@ usage( FILE* _fout ) {
         }
         fprintf( _fout, "%s\n", msgs[i] );
     }
-    printReleaseInfo( "iput" );
+    char name[] =  "iput";
```
[the remark this answers is not in the document]
Comment (korydraughn) [reply src/iput.cpp]: Done.

**Comment by korydraughn** [comment]
Squashed w/o pounds.

**Review by alanking — APPROVED** [review]
\# it

**Comment by korydraughn** [comment]
#'d

**Pull request merged**